### DISCUSSION TITLE
Is there a way to get all cells which references other cells?

### DISCUSSION THREAD

**Discussion by nathipg** [handsontable/hyperformula#820] [General]
Hello!

I'm migrating from other plugin and I'm replacing some code to use hyperformula.

Is there a way to get all cells which references other cells?
In the old plugin I could do it like this:

`formulaPlugin.sheet.matrix.cellReferences`

So it would return a array with cells like that:

![image](https://user-images.githubusercontent.com/14853904/134501885-df453de4-a6b2-4c00-8897-9218ced8dfe2.png)

Where row and column references the cell reference, and rowOffset and columnOffset references to the cell which is making the reference.

Hyperformula has something similar to that?

**Comment by wojciechczerniak** [comment]
[getCellDependents](https://handsontable.github.io/hyperformula/api/classes/hyperformula.html#getcelldependents) and [getCellPrecedents](https://handsontable.github.io/hyperformula/api/classes/hyperformula.html#getcellprecedents) are not what you're looking for?

**Reply by nathipg** [reply]
This is even better than what I asked for, thank you so much!
Sorry about the question, I must have miss it when I read the docs
